packetbeat/protos/dhcp/dhcp.go

```diff
@@ -0,0 +1,198 @@
+package dhcp
+
+import (
+	"bytes"
+	"encoding/hex"
+	"net"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/packetbeat/protos"
+)
+
+var hardwareTypes = map[int]string{
+	1:  "ethernet",
+	6:  "ieee_802",
+	7:  "arcnet",
+	11: "localtalk",
+	12: "localnet",
+	14: "smds",
+	15: "frame_relay",
+	16: "atm",
+	17: "hdlc",
+	18: "fibre_channel",
+	19: "atm",
+	20: "serial_line",
+}
+
+var messageTypes = []string{
+	"DHCPDISCOVER",
+	"DHCPOFFER",
+	"DHCPREQUEST",
+	"DHCPDECLINE",
+	"DHCPACK",
+	"DHCPNAK",
+	"DHCPRELEASE",
+	"DHCPINFORM",
+}
+
+type dhcpPlugin struct {
+	ports   []int
+	results protos.Reporter
+}
+
+func (dhcp *dhcpPlugin) init(results protos.Reporter, config *dhcpConfig) {
+	dhcp.ports = config.Ports
+	dhcp.results = results
+}
+
+func (dhcp *dhcpPlugin) GetPorts() []int {
+	return dhcp.ports
+}
+
+func payloadToRows(payload []byte) [][]byte {
+	result := make([][]byte, 0)
+	var idx = 0
+	for idx < len(payload) {
+		result = append(result, []byte{payload[idx], payload[idx+1], payload[idx+2], payload[idx+3]})
+		idx += 4
+	}
+	return result
+}
+
+func bytesToIPv4(bytes []byte) string {
+	ip := net.IPv4(bytes[0], bytes[1], bytes[2], bytes[3])
+	str := ip.String()
+	if str == "0.0.0.0" {
+		return ""
+	}
+	return str
+}
+
+func bytesToHardwareAddress(bytes []byte, length int) string {
+	return net.HardwareAddr(bytes[0:length]).String()
+}
+
+func combineRows(rows [][]byte, start int, length int) []byte {
+	combined := make([]byte, 0)
+	end := start + length
+	var idx = start
+	for idx < end {
+		combined = append(combined, rows[idx]...)
+		idx++
+	}
+	return combined
+}
+
+func trimNullBytes(b []byte) []byte {
+	index := bytes.Index(b, []byte{0})
+	return b[0:index]
+}
+
+func extractOptions(payload []byte) []byte {
+	result := make([]byte, 0)
+	var idx = 240
+	for idx < len(payload) {
+		result = append(result, payload[idx])
+		idx++
+	}
+	return result
+}
```

Comment:
Same as just using payload[240:], don't need a function for this unless there is a point in copying the slice, which you can do with copy(...)